engines/sci/engine/kgraphics32.cpp

```diff
@@ -707,66 +743,64 @@ reg_t kPalVaryUnknown2(EngineState *s, int argc, reg_t *argv) {
 	return kStub(s, argc, argv);
 }
 
+enum {
+	kSetCycle = 0,
+	kDoCycle = 1,
+	kCyclePause = 2,
+	kCycleOn = 3,
+	kCycleOff = 4
+};
+
 reg_t kPalCycle(EngineState *s, int argc, reg_t *argv) {
 	// Examples: GK1 room 480 (Bayou ritual), LSL6 room 100 (title screen)
 
 	switch (argv[0].toUint16()) {
-	case 0: {	// Palette animation initialization
-		// 3 or 4 extra params
-		// Case 1 sends fromColor and speed again, so we don't need them here.
-		// Only toColor is stored
-		//uint16 fromColor = argv[1].toUint16();
-		s->_palCycleToColor = argv[2].toUint16();
-		//uint16 speed = argv[3].toUint16();
-
-		// Invalidate the picture, so that the palette steps calls (case 1
-		// below) can update its palette without it being overwritten by the
-		// view/picture palettes.
-		g_sci->_gfxScreen->_picNotValid = 1;
-
-		// TODO: The fourth optional parameter is an unknown integer, and is 0 by default
-		if (argc == 5) {
-			// When this variant is used, picNotValid doesn't seem to be set
-			// (e.g. GK1 room 480). In this case, the animation step calls are
-			// not made, so perhaps this signifies the palette cycling steps
-			// to make.
-			// GK1 sets this to 6 (6 palette steps?)
-			g_sci->_gfxScreen->_picNotValid = 0;
-		}
-		kStub(s, argc, argv);
+	case kSetCycle: {
+		uint16 fromColor = argv[1].toUint16();
+		uint16 toColor = argv[2].toUint16();
+		int16 direction = argv[3].toSint16();
+		uint16 delay = (argc == 4 ? 0 : argv[4].toUint16());
```

Comment:
Since these 4 are the most variables that are actually used in this switch, it would be better to define them outside the switch, to avoid enclosing all of the cases in curly brackets